system/include/emscripten/thread_utils.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <emscripten.h>
+#include <emscripten/threading.h>
+#include <pthread.h>
+
+#include <functional>
+#include <thread>
+#include <utility>
+
+namespace emscripten {
+
+// Helper class for generic sync-to-async conversion. Creating an instance of
+// this class will spin up a pthread. You can then call invoke() to run code
+// on that pthread. The work done on the pthread receives a callback method
+// which lets you indicate when it finished working. The call to invoke() is
+// synchronous, while the work done on the other thread can be asynchronous,
+// which allows bridging async JS APIs to sync C++ code.
+//
+// This can be useful if you are in a location where blocking is possible (like
+// a thread, or when using PROXY_TO_PTHREAD), but you have code that is hard to
+// refactor to be async, but that requires some async operation (like waiting
+// for a JS event).
+class sync_to_async {
```

Comment:
Isn't the coding style to use CamelCase for class names?

Reply:
Fixed.